pkg/client/errors.go

```diff
@@ -24,12 +24,25 @@ import (
 
 // Errors related to Client connection and health check
 var (
+
+	// ErrIllegalArguments indicates illegal values of function argument
 	ErrIllegalArguments = errors.New("illegal arguments")
 
-	ErrAlreadyConnected   = errors.New("already connected")
-	ErrNotConnected       = errors.New("not connected")
-	ErrHealthCheckFailed  = errors.New("health check failed")
+	// ErrAlreadyConnected is used when trying to connect using already connected client
```

Comment:
ErrAlreadyConnected is used when trying to connect using an existing connected client? wdyt?

Reply:
Changed to `ErrAlreadyConnected is used when trying to establish a new connection with a client that is already connected` in 3b17be074cc700645a5d1888d152b0f76babe937 - let me know if it sounds better.

Reply:
lgtm!